src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -116,4 +113,71 @@ public function action(Message $message): void
             throw new Exception('Error sending message: ' . $error->getMessage(), 500);
         }
     }
+
+    protected static function createAdapterFromDSN(DSN $dsn): SMSAdapter
+    {
+        $from = empty($dsn->getParam('from', '')) ? null : $dsn->getParam('from', '');
```

Comment:
I thought `from` is always fetched from the environment variable? How is it populated in this property in the DSN ? Im not sure this is documented

Reply:
By default, it is the _APP_SMS_FROM variable.

If you have ?from= set in the DSN, this overrides it.

Will document this.